src/core/client/auth/containers/SignUpContainer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+import { BadUserInputError } from "talk-framework/lib/errors";
+import SignUp, { SignUpForm } from "../components/SignUp";
+
+import {
+  SetViewMutation,
+  SignUpMutation,
+  withSetViewMutation,
+  withSignUpMutation,
+} from "../mutations";
+
+interface SignUpContainerProps {
+  signUp: SignUpMutation;
+  setView: SetViewMutation;
+}
+
+interface SignUpContainerState {
+  error: string | null;
+}
+
+export type View = "SIGN_IN";
+
+class SignUpContainer extends Component<
+  SignUpContainerProps,
+  SignUpContainerState
+> {
+  public state = { error: "" };
```

Comment:
We don't need an error state, final form has support for this: See https://github.com/final-form/final-form#error-any-1

Reply:
Above